lib/http.js

```diff
@@ -284,8 +284,15 @@ module.exports = function (chai, _) {
     new Assertion(this._obj).to.redirect;
 
     if(redirects && redirects.length) {
+      var hasRedirected;
+
+      if (Object.prototype.toString.call(destination) === '[object RegExp]') {
+        hasRedirected = redirects.reduce((flag, redirect) => flag || destination.test(redirect), false);
```

Comment:
`reduce` will keep iterating over all members of the array; a faster way of iterating would be to use `some` which escapes as soon as it hits a `true`:

```suggestion
        hasRedirected = redirects.some(redirect => destination.test(redirect));
```

`some()` only ever returns a boolean: the boolean is `true` if _one element matches the test_, and `false` if _no elements match the test_

Reply:
Nice, I wasn't aware of this method. I'll push up the changes to use it, thanks.